worker/runfmusite/runFMUSite.py

```diff
@@ -0,0 +1,226 @@
+#  disclaimer.
+#
+#  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+#  disclaimer in the documentation and/or other materials provided with the distribution.
+#
+#  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+#  derived from this software without specific prior written permission from the respective party.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+#  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+#  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+#  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+#  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+########################################################################################################################
+
+from __future__ import print_function
+import os
+import glob
+import boto3
+import tarfile
+import shutil
+import time
+from pymongo import MongoClient
+import sys
+import subprocess
+import logging
+import re
+from datetime import date, datetime, timedelta
+import pytz
+import calendar
+import traceback
+from dateutil.parser import parse
+from pyfmi import load_fmu
+import copy
+import testcase
+import json
+
+
+def get_tag_data(tag_filepath):
+    '''
+     Purpose: retrieving the haystack tagged data, 
+              which are model-exchange variables in FMU. 
+     Inputs:  a json file containing all the tagged data.
+     Returns: a list for all the tagged data, with the tagging properties
+              like id, dis, site_ref, etc...
+    '''
+    with open( tag_filepath ) as json_data:
+        tag_data = json.load(json_data)
+        #print(" )))))) Hey tag data are: (((((( ",tag_data)
+
+    return tag_data
+
+
+def create_DisToID_dictionary(tag_filepath):
+    '''
+    Purpose: matching the haystack display-name and IDs
+    Inputs:  a json file containing all the tagged data
+    Returns: a dictionary mathching all display-names and IDs
+             a dictionary matching all inputs and IDs
+             a dictionary matching all outputs and IDs 
+    '''
+    dis_and_ID={}
+    inputs_and_ID = {}
+    outputs_and_ID = {}
+
+    tag_data = get_tag_data(tag_filepath)
+
+    for point in tag_data:
+        var_name = point['dis']
```

Comment:
Note that 'dis' will have :input and :output appended and that will not match the fmu keys.

Reply:
resolved.